### PR TITLE
batteries-3.7.1

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
supersedes https://github.com/ocaml/opam-repository/pull/24327

File: packages/batteries/batteries.3.7.1/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+conflicts: ["base-effects" "ocaml-option-no-flat-float-array"]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
```
Comment (mseri) [inline packages/batteries/batteries.3.7.1/opam:36]:
```suggestion
    jobs {!with-test}
    1 {with-test}
```
Should we just remove parallelism in the tests?

This should prevent
```
# (cd _build/default/src && /home/opam/.opam/4.14/bin/qtest extract --preamble-file ../qtest/qtest_preamble.ml --quiet extlib.ml batteriesPrint.ml batteriesExceptionless.ml batteriesConfig.ml batteries.ml batVect.ml batUref.ml batUnix.ml batUnit.ml batUTF8.ml batUChar.ml batTuple.ml batText.ml batSys.ml batSubstring.ml batString.ml batStream.ml batStack.ml batSplay.ml batSet.ml batSeq.ml batScanf.ml batReturn.ml batResult.ml batRefList.ml batRef.ml batRandom.ml batQueue.ml batPrintf.ml batPrintexc.ml batPervasives.ml batPathGen.ml batParserCo.ml batOrd.ml batOption.ml batOptParse.ml batOpaqueInnerSys.ml batNumber.ml batNum.ml batNativeint.ml batMultiPMap.ml batMultiMap.ml batMarshal.ml batMap.ml batLogger.ml batLog.ml batList.ml batLexing.ml batLazyList.ml batInterfaces.ml batInt64.ml batInt32.ml batInt.ml batInnerWeaktbl.ml batInnerShuffle.ml batInnerPervasives.ml batInnerIO.ml batISet.ml batIO.ml batIMap.ml batHeap.ml batHashtbl.ml batHashcons.ml batGlobal.ml batGenlex.ml batGc.ml batFormat.ml batFloat.ml batFingerTree.ml batFilename.ml batFile.ml batEnum.ml batEither.ml batDynArray.ml batDllist.ml batDigest.ml batDeque.ml batConcurrent.ml batConcreteQueue.ml batComplex.ml batCharParser.ml batChar.ml batCache.ml batBytesCompat.ml batBytes.ml batBuffer.ml batBounded.ml batBool.ml batBitSet.ml batBigarray.ml batBig_int.ml batBase64.ml batAvlTree.ml batArray.ml batVect.mli batUref.mli batUnix.mli batUnit.mli batUTF8.mli batUChar.mli batTuple.mli batText.mli batSys.mli batSubstring.mli batString.mli batStream.mli batStack.mli batSplay.mli batSet.mli batSeq.mli batScanf.mli batReturn.mli batResult.mli batRefList.mli batRef.mli batRandom.mli batQueue.mli batPrintf.mli batPrintexc.mli batPervasives.mli batPathGen.mli batParserCo.mli batOrd.mli batOption.mli batOptParse.mli batNumber.mli batNum.mli batNativeint.mli batMultiPMap.mli batMultiMap.mli batMarshal.mli batMap.mli batLogger.mli batLog.mli batList.mli batLexing.mli batLazyList.mli batInterfaces.mli batInt64.mli batInt32.mli batInt.mli batInnerWeaktbl.mli batInnerIO.mli batISet.mli batIO.mli batIMap.mli batHeap.mli batHashtbl.mli batHashcons.mli batGlobal.mli batGenlex.mli batGc.mli batFormat.mli batFloat.mli batFingerTree.mli batFilename.mli batFile.mli batEnum.mli batEither.mli batDynArray.mli batDllist.mli batDigest.mli batDeque.mli batConcurrent.mli batComplex.mli batCharParser.mli batChar.mli batCache.mli batBytes.mli batBuffer.mli batBounded.mli batBool.mli batBitSet.mli batBigarray.mli batBig_int.mli batBase64.mli batAvlTree.mli batArray.mli) > /opam-tmp/build_c89d1f_dune/dune-pipe-action-_ce1adf_.stdout
# Fatal error: exception Sys_error("batteriesConfig.ml: No such file or directory")
```

**Comment by UnixJunkie** [comment]
I am completely OK to remove parallelization of the tests.
This is starting to be fairly annoying if the CI tests don't behave as developers' computers...

**Comment by UnixJunkie** [comment]
I was not even happy that I had to retag batteries: it works, we ship.

**Comment by UnixJunkie** [comment]
To me, if the package compiles and install successfully, this is enough in terms of package testing.
Bonus points if we can known if it breaks reverse dependencies.
The test suite is a developers' thing (make test -> dune runtest; I am not even sure if it parallelize things).

File: packages/batteries/batteries.3.7.1/opam [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs {!with-test}
+    1 {with-test}
+    "@install"
+    "@runtest" {with-test}
```
Comment (mseri) [inline packages/batteries/batteries.3.7.1/opam:39]:
```suggestion
```
The fix worked for this release. I think there is value in testing on multiple archs, oses and versions of the compiler, but if next time you don't want to bother you can disable the tests in the releases by removing the corresponding call from the opam file

**Comment by mseri** [comment]
I am re-running some revdeps, but the failure were infrastructural so I plan to merge soon

File: packages/batteries/batteries.3.7.1/opam [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs {!with-test}
+    1 {with-test}
```
Comment (mseri) [inline packages/batteries/batteries.3.7.1/opam:37]:
```suggestion
    "1" {with-test}
```
Due to 
```
2023-09-14 14:06.13: Error in batteries.3.7.1:              error  3: File format error in 'build' at line 37, column 4: expected ident-or-string
```

File: packages/batteries/batteries.3.7.1/opam [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+synopsis: "A community-maintained standard library extension"
+maintainer: [
+  "Cedric Cellier <rixed@happyleptic.org>"
+  "Francois Berenger <unixjunkie@sdf.org>"
+  "Gabriel Scherer <gabriel.scherer@gmail.com>"
+  "Thibault Suzanne <thi.suzanne@gmail.com>"
+  "Simmo Saan <simmo.saan@gmail.com>"
+]
+authors: ["OCaml batteries-included team"]
+license: "LGPL-2.1-or-later with OCaml-LGPL-linking-exception"
```
Comment (mseri) [inline packages/batteries/batteries.3.7.1/opam:11]:
```suggestion
license: "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
```

**Comment by mseri** [comment]
Now also the lint is happy. The remaining failures are unrelated

**Pull request merged**